src/App.tsx

```diff
@@ -164,14 +165,19 @@ const App: React.FC = () => {
                     setIsLanguageModal={setIsLanguageModal}
                     isEditModal={isEditModal}
                     setIsEditModal={setIsEditModal}
+                    isMarkModal={isMarkModal}
+                    setIsMarkModal={setIsMarkModal}
                   />
                 </Route>
 
                 <Route
                   exact
                   path="/details"
                 >
-                  <TabDetails />
+                  <TabDetails
+                    isMarkModal={isMarkModal}
```

Comment:
I'm not sure about `MarkModal` inside `TabDetails`
Maybe just leave "There are no cycles yet" and remove `MarkModal`?

Reply:
Ok, I agree. I tested it and there were a lot of problems with this modal. (for example, back button, incorrect opening, etc.)